### ISSUE TITLE
Formatter fails to parse enums with mixin-only members

### ISSUE THREAD

**Issue by kubukoz** [disneystreaming/smithy-translate#171]
## Reproduction:

1. Create a file like `a.smithy`

```smithy
$version: "2"
namespace demo

@mixin enum X {
  A
}

enum Y with [X] {}
```

2. Run formatter

```
cs launch --channel https://disneystreaming.github.io/coursier.json smithytranslate --ttl 0 -- format ./a.smithy
```

3. See error

```
No files were formatted

The following files were not formatted:

a.smithy
for the following reasons
unable to parse the Smithy file: a.smithy for the following reason: failed at offset 53 with error: ...
}

enum Y with [X] {}
^
expectation:
* must end the string
```

## Expected behavior

The file parses and formats.

## Alternatives

There's a workaround, but it's not pretty: you can duplicate a member that's already inherited, that way the formatter sees that there's at least one enum case.

**Comment by daddykotex** [comment]
now that the smithy-cli has a formatter, my opinion is that we should focus else where. we have not discussed it yet, but I'll bring it up tomorrow. It still not perfect because I see failure in our internal api-registry, but eh

Release that introduced it: https://github.com/smithy-lang/smithy/releases/tag/1.33.0

**Comment by kubukoz** [comment]
> now that the smithy-cli has a formatter, my opinion is that we should focus else where.

yeah, I'm aware of that and I'm hoping to start using that formatter when it's added to the LSP. Having said that, I have no clue when it'll happen but I do have users hitting this 😅 

the workaround is straightforward enough that we can live with it for some time, though.

**Comment by daddykotex** [comment]
I'll keep that in mind and have a look as soon as I can :)

**Comment by daddykotex** [comment]
Looking at it

**Comment by daddykotex** [comment]
Released in 0.3.10
